### PR TITLE
fix(core): fix insertContentAt keeping new lines in html content

### LINKED ISSUE #2720: The insertContent command adds unnecessary blank paragraphs with HTML as value
### What’s the bug you are facing?

The `insertCommand` seems to be adding unnecessary blank paragraphs to the editor when newlines are found between HTML tags.

### Which browser was this experienced in? Are any special extensions installed?

n/a

### How can we reproduce the bug on our side?

Take the following HTML as an example:

```js
const htmlInput= `<h1><a href="https://tiptap.dev/">Tiptap</a></h1>
<p><strong>Hello World</strong></p>`
```

And then call `editor.commands.insertCommand(htmlInput)`. This is what gets rendered by Tiptap:

```html
<div
    translate="no"
    role="textbox"
    aria-readonly="false"
    aria-multiline="true"
    class="ProseMirror"
    tabindex="0"
    contenteditable="true"
>
    <p><br class="ProseMirror-trailingBreak" /></p>
    <h1>
        <a target="_blank" rel="noopener noreferrer nofollow" href="https://tiptap.dev">Tiptap</a>
    </h1>
    <p></p>
    <p><strong>Hello World</strong></p>
</div>
```

As you can see, there's one paragraph before and after the header that shouldn't be there.

### Can you provide a CodeSandbox?

_No response_

### What did you expect to happen?

Based on the example above, Tiptpa should've rendered this:

```html
<div
    translate="no"
    role="textbox"
    aria-readonly="false"
    aria-multiline="true"
    class="ProseMirror"
    tabindex="0"
    contenteditable="true"
>
    <h1>
        <a target="_blank" rel="noopener noreferrer nofollow" href="https://tiptap.dev">Tiptap</a>
    </h1>
    <p><strong>Hello World</strong></p>
</div>
```

### Anything to add? (optional)

_No response_

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [X] Yes, I’m a sponsor. 💖 _(on behalf of Doist)_

### PR DESCRIPTION
**Pull request by bdbch**
## Please describe your changes

This PR fixes the issue described in #2720 which was caused by keeping new lines in string content and using the DOMParser which will add breaks for new lines too.

## How did you accomplish your changes

I strip out `\n` content from the content string so new lines are interpretet by the DOMParser only.

## How have you tested your changes

I created a demo where this can be tested. (See [here](https://deploy-preview-4465--tiptap-embed.netlify.app/src/issues/2720/react/))

## How can we verify your changes

Test in the demo above

## Checklist

- [x] The changes are not breaking the editor
- [x] Added tests where possible
- [x] Followed the guidelines
- [x] Fixed linting issues

## Related issues

- fixes #2720

### CODE REVIEW

File: packages/core/src/commands/insertContentAt.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ export const insertContentAt: RawCommands['insertContentAt'] = (position, value,
         tr.insertText(value as string, from, to)
       }
     } else {
+      console.log(content)
```
Comment (rfgamaral) [inline packages/core/src/commands/insertContentAt.ts:92]: 🚨🚨🚨

Comment (bdbch) [reply packages/core/src/commands/insertContentAt.ts:92]: Thanks for the quick review!

**Comment by bdbch** [comment]
@svenadlung can you take a look too? self reviewing right now.

**Comment by bdbch** [comment]
Update: Found some more issues I'd have to take a look into.

**Comment by bdbch** [comment]
Okay - ready. @svenadlung if you have some time a review would be great.

**Comment by bdbch** [comment]
Ping @svenadlung

**Comment by svenadlung** [comment]
@bdbch Sorry for the delay. At the first moment it feels to me like something could break of the change. Please give me some more time for a review. Probably until the end of next week due to public holidays.

**Comment by gdfreitas** [comment]
We're also facing this issue, do we have any updates? @svenadlung

**Comment by kaiserkiwi** [comment]
Any progress here?

**Comment by bdbch** [comment]
@svenadlung should we include this in our next release or should we postpone it for a later release?

**Comment by murisceman** [comment]
This fix breaks `<pre><code>foo\nbar</code></pre>`.

**Comment by bdbch** [comment]
working on a fix @murisceman

**Comment by bdbch** [comment]
PR is here: https://github.com/ueberdosis/tiptap/pull/4767

**Pull request merged**